src/coreclr/src/jit/lowerarmarch.cpp

```diff
@@ -914,7 +914,29 @@ void Lowering::ContainCheckHWIntrinsic(GenTreeHWIntrinsic* node)
                 }
             }
             break;
+        case NI_Vector64_CreateScalarUnsafe:
+        case NI_Vector128_CreateScalarUnsafe:
+            if (intrin.op1->IsCnsIntOrI())
+            {
+                const ssize_t dataValue = intrin.op1->AsIntCon()->gtIconVal;
+
+                if (comp->GetEmitter()->emitIns_valid_imm_for_movi(dataValue, emitActualTypeSize(intrin.baseType)))
```

Comment:
Does this also check that op1 can be encoded in mvni?

Reply:
Yes, it does [here](https://github.com/dotnet/runtime/blob/master/src/coreclr/src/jit/emitarm64.cpp#L2253).